app/assets/javascripts/home.coffee

```diff
@@ -26,19 +26,46 @@ set_coords_by_geolocation = (position) ->
   set_coords(position.coords.accuracy, position.coords.latitude, position.coords.longitude)
 
 block_callback = (err) ->
-  location_error()
+  if $('#location_geolocation').prop('checked')
+    $('#location_button').prop('innerHTML', 'Get My Location')
+
+  $('#error-geolocation').modal('hide')
+  $('#error-position_unavailable').modal('hide')
+
+  if err.code == err.POSITION_UNAVAILABLE && is_safari()
+    $('#error-position_unavailable').modal('show')
+  else 
+    $('#error-geolocation').modal('show')
 
   Sentry.setExtra("error_code", err.code)
   Sentry.setExtra("error_message", err.message)
   Sentry.captureException(err)
 
+is_safari = ->
+  ua = navigator.userAgent.toLowerCase()
+  if ua.indexOf('safari') != -1
+    if ua.indexOf('chrome') > -1
+      return false
+    return true
+  return false
```

Comment:
We should use consistent tests for return values of `.indexOf()`, either `!= -1` or `> -1`, but not both. (I'd like `.contains()` even better, but we'd need to check browser usage before doing that)

Also, this can all be simplified to:

```javascript
if ua.indexOf('safari') != -1
  return ua.indexOf('chrome') == -1
return false
```

And even further:

```javascript
return (ua.indexOf('safari') != -1) && (ua.indexOf('chrome') == -1)
```

Reply:
**Edit:** Whoops, I referenced a non-existent method. It would be `<array>.includes()`, not `<array>.contains()`, if we go that route.

Reply:
Oh, sorry! I think I communicated that in a confusing way, since I didn't include the entire original block in my quote. What I meant was that that line could replace the *whole* nested if-else block. That line by itself essentially says: "Return `true` only if `'safari'` is in the ua string and `'chrome'` *isn't* in the string, otherwise return `false`."

...so the entire `is_safari` function would just be:

```javascript
is_safari = ->
    ua = navigator.userAgent.toLowerCase()
￼	  return (ua.indexOf('safari') != -1) && (ua.indexOf('chrome') == -1)
```

Just that one adjustment and you'll be good to go!